app/controllers/verify/sessions_controller.rb

```diff
@@ -14,18 +15,16 @@ def new
     end
 
     def create
-      idv_session.params.merge!(profile_params)
-      submit_profile
+      process_step
     end
 
     private
 
-    def submit_profile
-      result = idv_profile_form.submit(profile_params)
-      analytics.track_event(Analytics::IDV_BASIC_INFO_SUBMITTED, result)
-
-      if result[:success]
+    def process_step
+      if step.complete
         redirect_to verify_finance_path
+      elsif step.attempts_exceeded?
+        redirect_to verify_fail_path
```

Comment:
I take back that last part about removing the code from the Step class. We need that for analytics to track when and where the attempts were first exceeded.